### DISCUSSION TITLE
Berry and SK6812

### DISCUSSION THREAD

**Discussion by martin072** [arendst/Tasmota#19146] [Q&A]
I am trying to play with berry and a led strip, but when trying to configure as follows, the device crashes 

```self.strip = Leds(11, 22, Leds.SK6812_GRBW)```

```Guru Meditation Error: Core  1 panic'ed (LoadProhibited). Exception was unhandled.```

If I leave the module type out, i.e. 

```self.strip = Leds(11, 22)```

then it runs the script fine, but colors are not showing correct. 

any ideas?

**Comment by s-hadinger** [comment]
I'm afraid nobody ever tried using Leds.SK6812_GRBW. Let me look at it in the following days.

**Reply by martin072** [reply]
Much appreciated, no rush.

**Reply by s-hadinger** [reply]
I cannot reproduce. Trying `Leds(11, 22, Leds.SK6812_GRBW)` does not crash.

Do you have more to share?

**Reply by martin072** [reply]
own compiled firmware (based on esp32 environment) from Mac M1 with Visual Code, using my own user_config_override. could it be that there is something interfering with certain combinations?

**Reply by s-hadinger** [reply]
I have also a MacM1 with VSCode, it has no impact. I doubt that the compilation options have any influence.

**Reply by sfromis** [reply]
With precompiled `tasmota32s3cdc-qio_opi.factory.bin`, I reproduced it:
`var pix = Leds(1, gpio.pin(gpio.WS2812,0), Leds.SK6812_GRBW, 1) pix.set_pixel_color(0,0xff0000) pix.show()`

`{"Info3":{"RestartReason":{"Exception":28,"Reason":"LoadProhibited","EPC":"42090f5e","EXCVADDR":"00000004","CallChain":["42090f5b","4203f749","4203f873","42041005","42042052","42048f5a","4208bd62","420799c5","4203320a","420334b5","42034084","420aa1e1","4207dc19","420b7892","420bfe11","4202717b"]},"BootCount":30}}`

I've not checked if `qio_opi` is "most correct" for the ESP32-S3FH4R2 chip, but it has seemed to work fine with it.
Selection of a QIO binary was due to esptool.py saying:
> Flash type set in eFuse = quad (4 data lines)

Anyway, color order is bad on my board, red/green swapped, but the Berry setup has nothing but GRB.

**Reply by martin072** [reply]
I use the 'tasmota32', not a s3. see picture, its based on wroom32

<img width="469" alt="Screenshot 2023-07-21 at 14 30 58" src="https://github.com/arendst/Tasmota/assets/24524506/cee2ea9d-dcff-4862-a0d8-cada7301d370">

**Reply by sfromis** [reply]
Does not look hardware platform dependent, I'm also seeing it with precompiled dev binaries on C3 and S2.

**Comment by s-hadinger** [comment]
Fixed in #19166. It wad a stupid cut/paste mistake.

It doesn't crash anymore but I still can't test it.

**Reply by sfromis** [reply]
Indeed, my test case no longer crashed.

Still getting wrong color order, which surprised me as the schematic says "WS2812B-0807". Of course nothing to do with this issue, and I see exactly the same (red/green swapped) with both available led models.
